lib/utils.js

```diff
@@ -2,12 +2,13 @@
 
 var Promise = require('pouchdb-promise');
 var urlJoin = require('url-join');
+var parse = require('url-parse');
 
 function getBaseUrl(db) {
   if (typeof db.getUrl === 'function') { // pouchdb pre-6.0.0
-    return db.getUrl().replace(/\/[^\/]+\/?$/, '');
+    return parse(db.getUrl()).origin;
   } else { // pouchdb post-6.0.0
-    return db.name.replace(/\/[^\/]+\/?$/, '');
+    return parse(db.__opts && db.__opts.prefix || db.name).origin;
```

Comment:
You added a `db.__opts && db.__opts.prefix` case. What does it add compared to `db.name` ?